### ISSUE TITLE
Carry role unable to dismantle

### ISSUE THREAD

**Issue by DarkDestry** [TooAngel/screeps#467]
Just getting into the bot and trying to figure out the code behind the amazing AI

Regarding [#151 : Let carries dismantle structures on path](https://github.com/TooAngel/screeps/pull/151)
I am currently facing an issue as such
![image](https://user-images.githubusercontent.com/3728955/31586919-c3a1f6b2-b20a-11e7-98b2-838e890fcf28.png)

The dismantle code is throwing `ERR_NO_BODYPART` and thus causing all carrys to jam when facing an enemy structure.

What was the intended result when the code was merged? I am a rather new player so unless the `ERR_NO_BODYPART` was added post merge, this is currently broken.

**Comment by DarkDestry** [comment]
I looked a little deeper and found that `carry` role only gets `W` after hitting a energy capacity requirement. Problem being when reclaiming used/abandoned rooms, this cannot be satisfied as most of the existing structures will clog up the path causing the room to be blocked.

**Comment by TooAngel** [comment]
The idea behind the dismantle is:
If the AI handles external rooms the creeps walk on fixed paths, which ignore structures. Usually `structurers` should be spawned to clear out the way, but sometimes there can be still structures. So all (most of the) creeps should clear out their way.

In your case I guess it is your own room, eventually the storage will be destroyed by `checkWrongStructures`. Tbh I'm not sure about the shape of that part, I saw some cases where it took really long (so long that the issue was fixed manually) before certain structures were destroyed.
If that is your case you better of checking why the structure was not cleaned by your own room.
